src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -42,18 +42,12 @@
  *
  * @apiNote
  * To release resources used by this stream {@link #close} should be called
- * directly or by try-with-resources. Subclasses are responsible for the cleanup
- * of resources acquired by the subclass.
- * Subclasses that override {@link #finalize} in order to perform cleanup
- * should be modified to use alternative cleanup mechanisms such as
- * {@link java.lang.ref.Cleaner} and remove the overriding {@code finalize} method.
+ * directly or by try-with-resources.
  *
  * @implSpec
- * If this FileInputStream has been subclassed and the {@link #close}
- * method has been overridden, the {@link #close} method will be
- * called when the FileInputStream is unreachable.
- * Otherwise, it is implementation specific how the resource cleanup described in
- * {@link #close} is performed.
+ * Subclasses are responsible for the cleanup of resources acquired by the subclass.
+ * Subclasses requiring that resource cleanup take place after a stream becomes
+ * unreachable should use the {@link java.lang.ref.Cleaner} mechanism.
```

Comment:
This is in the style of an API note for implementors so implSpec is appropriate here.

Reply:
Perhaps add "or other mechanisms" or similar advice.  
There should be some room to use other kinds of queues or polling to do the cleanup.
For example, there are some maps that check for cleanups when doing other operations.